solarforecastarbiter/validation/validator.py

```diff
@@ -563,16 +563,73 @@ def check_irradiance_day_night(solar_zenith, max_zenith=87):
     Parameters
     ----------
     solar_zenith : Series
-        Solar zenith angle in degrees
-    max_zenith : maximum zenith angle for a daylight time
+        Solar zenith angle in degrees.
+    max_zenith : float
+        Maximum zenith angle for a daylight time.
 
     Returns
     -------
     flags : Series
         True when solar zenith is less than max_zenith.
     """
-    flags = _check_limits(solar_zenith, ub=max_zenith)
-    return flags
+    # True = daytime. False = nighttime.
+    daytime = _check_limits(solar_zenith, ub=max_zenith)
+    return daytime
+
+
+@mask_flags('NIGHTTIME')
+def check_irradiance_day_night_interval(
+        solar_zenith, closed, interval_length,
+        solar_zenith_interval_length=None, fraction_of_interval=0.1,
+        max_zenith=87):
+    """Check for day/night periods based on solar zenith.
+
+    Interval average data may be analyzed by supplying higher resolution
+    solar zenith data and parameters that describe the desired intervals.
+
+    Parameters
+    ----------
+    solar_zenith : Series
+        Solar zenith angle in degrees.
+    closed : {'left', 'right'}
+        None for instantaneous data, 'left' for interval beginning labeled
+        data, 'right' for interval ending labeled data.
+    interval_length : Timedelta
+        Interval length to resample day/night periods to.
+    solar_zenith_interval_length : None or Timedelta
+        If None, attempt to infer.
+        Required if solar_zenith contains gaps and closed is not None.
+    fraction_of_interval : float
+        The fraction of the points in an interval that must be daytime
+        in order to mark the interval as daytime.
+    max_zenith : float
+        Maximum zenith angle for a daylight time.
+
+    Returns
+    -------
+    flags : Series
+        True when sufficient points within an interval are less than
+        max_zenith. Index conforms to solar_zenith resampled to
+        interval_length.
+    """
+    # True = daytime. False = nighttime.
+    daytime = _check_limits(solar_zenith, ub=max_zenith)
+    # number of daytime minutes within the interval
+    daytime_sum = daytime.resample(
+        interval_length, closed=closed, label=closed
+    ).sum()
+    # if not provided, try to interval length for normalization.
+    # this will fail if the index has gaps.
```

Comment:
fail w/ what? an error? or will an error be raised when dividing by none below. Either way should add to docstring

Reply:
Ah I erroneously thought `warn=False` meant that it would raise an exception. Now the validator function will do it.